cpp/arcticdb/version/version_store_api.cpp

```diff
@@ -371,10 +371,8 @@ void PythonVersionStore::add_to_snapshot(
         retained_keys.emplace_back(std::move(key));
 
     std::sort(std::begin(retained_keys), std::end(retained_keys));
-    if(variant_key_type(snap_key) == KeyType::SNAPSHOT_REF && cfg().write_options().delayed_deletes()) {
-        tombstone_snapshot(store(), to_ref(snap_key), std::move(snap_segment), version_map()->log_changes());
-    } else {
-        delete_trees_responsibly(deleted_keys, get_master_snapshots_map(store()), snap_name).get();
+    if(variant_key_type(snap_key) != KeyType::SNAPSHOT_REF || !cfg().write_options().delayed_deletes()) {
```

Comment:
I know it was here already, but this check,

```
variant_key_type(snap_key) != KeyType::SNAPSHOT_REF
```

is really confusing. Don't we already know what type `snap_key` is??

Reply:
I had the same question when I was working on it. My guess is for backward compatibility with `KeyType::SNAPSHOT`:
https://github.com/man-group/ArcticDB/blob/3adbed459f6046d79b3a61264e2a5b5b62cc3ca8/cpp/arcticdb/version/snapshot.cpp#L214